src/core/polkadot.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+# Copyright 2025 Dwellir
+# See LICENSE file for licensing details.
+
+"""Functions for managing and interacting with the Polkadot workload.
+
+This module provides functionality for managing the Polkadot snap package,
+including installation, service management, and configuration.
+"""
+
+import logging
+from typing import Optional
+from subprocess import run, CalledProcessError
+from charms.operator_libs_linux.v2 import snap
```

Comment:
This lib is overlapping with much of the code you have in this class? Why import it if yoy aren't using the functionality but implement your own?